cmd/massa-wallet/thyra-plugin-wallet.go

```diff
@@ -44,11 +35,35 @@ func StartServer(app *fyne.App, port int) {
 	server := restapi.NewServer(massaWalletAPI)
 	server.ConfigureAPI()
 
-	// Set the port to listen on to the passed-in port
-	server.Port = port
+	ln, err := server.HTTPListener()
+	if err != nil {
+		log.Fatalln(err)
+	}
+	registerPlugin(ln)
 
 	if err := server.Serve(); err != nil {
-
 		log.Fatalln(err)
 	}
 }
+
+func registerPlugin(ln net.Listener) {
+	//nolint:gomnd
+	if len(os.Args) < 2 {
+		panic("this program must be run with correlation id argument!")
+	}
+
+	pluginID := os.Args[1]
```

Comment:
Not sure it makes sense to ask the enduser to define a plugin id when using standalone mode